motion/dp_controller/include/dp_controller/quaternion_pd_controller.h

```diff
@@ -17,63 +27,194 @@
 #include <iostream>
 using namespace Eigen;
 
+/**
+ * @brief A complete class for the nonlinear PID controller
+*/
 class QuaternionPdController
 {
 public:
+
+  /**
+   * @brief Controller class constructor
+   * 
+   * @param a     Diagonal value for derivative gain matrix
+   * @param b     Diagonal value for position gain matrix
+   * @param c     Orientation gain
+   * @param i     Scalar for orientation integral gain and diagonal value for integral gain matrix
+   * @param W     Weight of drone
+   * @param B     Buoyancy of drone
+   * @param r_G   Center of gravity, expressed in body frame
+   * @param r_B   Center of buoyancy, expressed in body frame
+   * 
+  */
   QuaternionPdController(double a, double b, double c, double i, double W, double B,
                          const Eigen::Vector3d &r_G, const Eigen::Vector3d &r_B);
 
+
+  /**
+   * @brief set the gains of the controller
+   * 
+   * @param a     Diagonal value for derivative gain matrix
+   * @param b     Diagonal value for position gain matrix
+   * @param c     Orientation gain
+   * @param i     Scalar for orientation integral gain and diagonal value for integral gain matrix
+   * 
+  */
   void setGains(double a, double b, double c, double i);
 
-  // Return restoring forces vector
+  /**
+   * @brief a getter for the restoring forces vector.
+   * 
+   * @see  restoringForceVector()
+   * 
+   * @return the restoring forces vector
+  */
   Eigen::Vector6d getRestoring(const Eigen::Quaterniond &q);
 
```

Comment:
For good measurement, a short explenation for @p q should be included

Reply:
Done :+1: